tests/functional/z_cluster/nodes/test_nodes_restart.py

```
@@ -205,25 +210,41 @@ def test_pv_provisioning_under_degraded_state_stop_provisioner_pod_node(
+        # Avoid rook operator's node, and NooBaa's node (may miss bucket_creation_timeout)
         rook_operator_pod = pod.get_operator_pods()[0]
         operator_node = pod.get_pod_node(rook_operator_pod)
-        if operator_node.get().get("metadata").get(
-            "name"
-        ) == provisioner_node.get().get("metadata").get("name"):
+        unsafe_nodes = {operator_node.name}
+        if operation == "create_resources":
+            try:
+                unsafe_nodes.update(get_app_pod_running_nodes(pod.get_noobaa_pods()))
+            except (CommandFailed, NotFoundError) as ex:
+                logger.warning(f"Could not determine NooBaa pod nodes: {ex}")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)